bpf/go_nethttp.h

```diff
@@ -10,30 +13,930 @@
+
+    connection_info_t *conn = bpf_map_lookup_elem(&ongoing_server_connections, &goroutine_addr);
+
+    if (conn) {
+        bpf_dbg_printk("Found existing server connection, parsing FD information for socket tuples, %llx", goroutine_addr);
+
+        void *fd_ptr = GO_PARAM1(ctx);
+        get_conn_info_from_fd(fd_ptr, conn); // ok to not check the result, we leave it as 0
+
+        //dbg_print_http_connection_info(conn);
+    }
+
+    return 0;
+}
+
+SEC("uprobe/connServeRet")
+int uprobe_connServeRet(struct pt_regs *ctx) {
+    bpf_dbg_printk("=== uprobe/proc http conn serve ret === ");
+    void *goroutine_addr = GOROUTINE_PTR(ctx);
+
+    bpf_map_delete_elem(&ongoing_server_connections, &goroutine_addr);
+
+    return 0;
+}
+
+SEC("uprobe/persistConnRoundTrip")
+int uprobe_persistConnRoundTrip(struct pt_regs *ctx) {
+    bpf_dbg_printk("=== uprobe/proc http persistConn roundTrip === ");
+    void *goroutine_addr = GOROUTINE_PTR(ctx);
+    bpf_dbg_printk("goroutine_addr %lx", goroutine_addr);
+
+    http_func_invocation_t *invocation = bpf_map_lookup_elem(&ongoing_http_client_requests, &goroutine_addr);
+    if (!invocation) {
+        bpf_dbg_printk("can't find invocation info for client call, this might be a bug");
+        return 0;
+    }
+
+    void *pc_ptr = GO_PARAM1(ctx);
+    if (pc_ptr) {
+        void *conn_conn_ptr = pc_ptr + 8 + pc_conn_pos; // embedded struct
+        void *tls_state = 0;
+        bpf_probe_read(&tls_state, sizeof(tls_state), (void *)(pc_ptr + pc_tls_pos)); // find tlsState
+        bpf_dbg_printk("conn_conn_ptr %llx, tls_state %llx", conn_conn_ptr, tls_state);
+
+        conn_conn_ptr = unwrap_tls_conn_info(conn_conn_ptr, tls_state);
+
+        if (conn_conn_ptr) {
+            void *conn_ptr = 0;
+            bpf_probe_read(&conn_ptr, sizeof(conn_ptr), (void *)(conn_conn_ptr + net_conn_pos)); // find conn
+            bpf_dbg_printk("conn_ptr %llx", conn_ptr);            
+            if (conn_ptr) {
+                connection_info_t conn = {0};
+                get_conn_info(conn_ptr, &conn); // initialized to 0, no need to check the result if we succeeded
+                u64 pid_tid = bpf_get_current_pid_tgid();
+                u32 pid = pid_from_pid_tgid(pid_tid);
+                tp_info_pid_t tp_p = {
+                    .pid = pid,
+                    .valid = 1,
+                };
+
+                tp_clone(&tp_p.tp, &invocation->tp);
+                tp_p.tp.ts = bpf_ktime_get_ns();
+                bpf_dbg_printk("storing trace_map info for black-box tracing");
+                bpf_map_update_elem(&ongoing_client_connections, &goroutine_addr, &conn, BPF_ANY);
+
+                // Must sort the connection info, this map is shared with kprobes which use sorted connection
+                // info always.
+                sort_connection_info(&conn);
+                bpf_map_update_elem(&trace_map, &conn, &tp_p, BPF_ANY);
+            }
+        }
+    }
+
+    return 0;
+}
+
+// SQL support
```

Comment:
Now that we are doing is combining, can we please split this code below into go_sql.h? We combined it because it was such small code and it was duplicating a lot of memory and in all go sql cases we always loaded something from what we probe on nethttp.